mod/gcforums/start.php

```diff
@@ -48,696 +53,670 @@ function gcforums_owner_block_menu($hook,$type,$return,$params) {
-										<th class='gcforums-td'>".get_recent_post($forum->guid)."</th>
-										<th class='gcforums-td'>".gcforums_forums_edit_options($forum->guid, $group_guid)."</th>
-									</tr>";
-				}
+		if (sizeof($forums) > 0) {
+				
+			$content .= "
+				<div class='forum-main-box'>
+					<div style='background: #e6e6e6; width:100%;' >
+						<div class='forum-header'>Forum
+							<div class='forum-information'>options</div>
+							<div class='forum-information'>total topics</div>
+							<div class='forum-information'>total posts</div>
+							<div class='forum-information'>recently posted</div>
+						</div>";
+
+			foreach ($forums as $forum) {
+					$total_topics = get_forums_statistics_information($forum->guid, TOTAL_TOPICS);
+					$total_posts = get_forums_statistics_information($forum->guid, TOTAL_POST);
+					$recent_post = get_forums_statistics_information($forum->guid, RECENT_POST);
+					$options = render_edit_options($forum->getGUID(), $group_entity->getGUID());
+
+					$hyperlink = "<a href='{$base_url}gcforums/view/{$forum->getGUID()}'><strong>{$forum->title}</strong></a>";
+					$admin_only = (elgg_is_admin_logged_in()) ? "(guid:{$forum->guid})" : "";
+
+					$content .= "
+						<div class='forum-info-header'>
+							<div class='forum-description'>{$hyperlink} {$admin_only}
+								<div class='forum-description-text'>{$forum->description}</div>
+							</div>
+							<div class='forum-options-edit'>{$options}</div>
+							<div class='forum-options'>{$total_topics}</div>
+							<div class='forum-options'>{$total_posts}</div>
+							<div class='forum-options'>{$recent_post}</div>
+						</div>";
 			}
+
+			$content .= "</div> </div> </p> <br/>";
 		}
 	}
 
-	$t_forum_category .= "</table>";
-    //putting the table together with the category out of the table - nick
-	return $t_forum_category;
-}
-
+	$title = $entity->title;
+	if (!$title) $title = elgg_echo('gcforum:heading:default_title');
 
+	$return['filter'] = '';
+	$return['title'] = $title;
+	$return['content'] = $content;
+	return $return;
+}
 
 
-/* Create list of options to modify forums
+/**
+ * TOTAL_POST : 1
+ * TOTAL_TOPICS : 2
+ * RECENT_POST : 3
  */
-function gcforums_forums_edit_options($object_guid,$group_guid) {
-	$edit_options = '';
-	if (elgg_is_logged_in() && ( elgg_get_logged_in_user_entity()->isAdmin() || get_entity($group_guid)->getOwnerGUID() == elgg_get_logged_in_user_guid())) {
+function get_forums_statistics_information($container_guid, $type) {
+	$dbprefix = elgg_get_config('dbprefix');
 	
-		$object_menu_items = array("New subforum", "New Posting", "Edit");
+	switch($type) {
+		case 1:
+			$query = "SELECT COUNT(r.guid_one) AS total
+				FROM {$dbprefix}entity_relationships r, {$dbprefix}entities e, {$dbprefix}entity_subtypes es
+				WHERE r.guid_one = e.guid AND e.subtype = es.id AND r.guid_two = {$container_guid} AND es.subtype = 'hjforumpost' AND (e.access_id = 1 OR e.access_id = 2)";
```

Comment:
pretty sure the mysql optimizer handles e.access_id IN(1,2) a bit better, though I doubt it'll be noticeable in this case with everything else being ANDs.